### DISCUSSION TITLE
SPI, master vs. slave

### DISCUSSION THREAD

**Discussion by chrisalbertson** [micropython/micropython#10442] [General]
I'm writiing a simple test on a Pico.  The pico has twp SPI blocks and two cores so my setup is this

Core0: uses SPI0 to write blocks of data
Core1: use SPI1 to read blocks of data

My logic analyzer shows data coming from SPO0 as expected.  My problem is that I can't seem to read any data from SPI1.

Questions:
1) I do not see where I can specify if the SPI block should act as a master or a slave. It seems to act as a master because I see output in MOSI.

2) I suspect you can not interconnect the two SCLK pins because both are trying to write.  But how to send SPI without doing so

3) Assuming that slave mode is not yet implemented in Micropython what do I have to do to read data from a master on the other SPI block?

My use case is to implement a slave mode SPI device that can be accessed from a Linux-based Pi4.

**Comment by robert-hh** [comment]
Slave mode is not supported yet by MicroPython. @dpgeorge has added on his almost infinite To-Do list the task of specifying the API. So sadly, it will take some time.

**Reply by bawanqin1987** [reply]
how long can i use the py case with spi slave?
